src/server/scripts/Outland/BlackTemple/boss_mother_shahraz.cpp

```diff
@@ -212,7 +212,19 @@ class spell_mother_shahraz_fatal_attraction : public SpellScript
 
     void SetDest(SpellDestination& dest)
     {
-        dest.Relocate(GetCaster()->GetRandomNearPosition(50.0f));
+        // Initialize a first destination
+        Position teleportDest = GetCaster()->GetRandomNearPosition(50.0f);
+
+        // Ensure that the destination is not too close to the caster.
+        // Add a check for LOS, to ensure to not be teleported under the map
+        while (!teleportDest.IsRadiusPositionValid(GetCaster(), teleportDest, 25.0f))
```

Comment:
I wouldn’t use this `while` loop here. It could result in an infinite loop or have unpredictable time complexity.

I don’t know much about this spell, but I assume it’s used by a single boss. If that’s the case, have you considered creating some safe rectangle(s) or circle(s) at the boss's location and simply finding a random point within the shape using a given radius?

Reply:
The boss can be tank in all the hall. So I can't create a specific area. I can also add a check to break the loop After a certain nomber of try. Otherwise I don't know how done this without "while loop" as we doesn't have specific waypoints.

Reply:
Why can't we use this rectangle area?
![369687081-5805168a-ba72-477a-a194-8a9554b4b964-2](https://github.com/user-attachments/assets/c6cdb920-cc1e-46c0-9648-a8c98ef24df3)
Or if looks not too good then maybe we can work with these 3 rectangles?
![2](https://github.com/user-attachments/assets/c984424a-eb7f-41d6-bae3-3f264bfd6f72)

Reply:
It's stay a hackfix if there are nothing define on sniffs. Thé first screen IS incorrect because you can bd TP outisde of this zone. And on thé second screen I Always see positions where you can be near to a wall. 

I think that blizzard does really a random but have a method to check walls position before to valid the destination. I've got another Idea. As my last commit, but instead to redone a new GetNearPosition if detect a wall, we relocate the final dest to the opposite of the wall (Can be done with few trigonometry formula)

Reply:
I don't know if this can help, but this error happen during Fatal Attraction teleport. I noted that when this error appear, Mother teleport the players near to the wall.
![invalid](https://github.com/user-attachments/assets/bd305c43-6d9f-4b3d-a652-29202a02e5f5)

Reply:
Thanks. I will look this today, I'm black from my professional travel

Reply:
Give me an advise when I can test it.